app/(root)/explore/page.tsx

```diff
@@ -0,0 +1,54 @@
+// ----------------------------------------------------------------
+
+import ExplorePostsList from '@/components/post/ExplorePostsList';
+import PostItemBadge from '@/components/post/PostItemBadge';
+import { POST_TYPES } from '@/constants';
+import { getAllPosts } from '@/lib/actions/post-actions';
+import { IPostsResponse } from '@/types/post';
+import { parseSearchParams } from '@/utils/params';
+
+// ----------------------------------------------------------------
+
+interface IExplorePageProps {
+  searchParams: {
+    page: string | string[] | undefined;
+    postType: string | string[] | undefined;
+  };
+}
+
+const ExplorePage: React.FC<IExplorePageProps> = async ({ searchParams }) => {
+  const page = Number(parseSearchParams(searchParams.page, '1'));
+
+  const response: IPostsResponse | undefined = await getAllPosts({
+    page: Number(page),
+    itemsPerPage: 10,
+  });
+
+  if (!response?.ok && response?.status !== 200) {
```

Comment:
```suggestion
  if (response?.status !== 200) {
```
This should be enough, as the "ok" field is calculated based on the status code anyway